app/Http/Controllers/Auth/AdmissionController.php

```diff
@@ -142,25 +145,30 @@ public function show(Application $application): View
     }
 
     /**
-     * Edit an application's note.
+     * Edit an application.
      * @param Request $request
      * @param Application $application
      * @return RedirectResponse
      * @throws AuthorizationException
      */
-    public function updateNote(Request $request, Application $application): RedirectResponse
+    public function update(Request $request, Application $application): RedirectResponse
     {
         $this->authorize('view', $application);
         if (user()->id == $application->user_id) {
             return redirect()->back()->with('error', 'You cannot modify the internal note of yourself.');
         }
-
-        $newStatus = $request->input('status_'.$application->user->id);
         if ($request->has('note')) {
+            $request->validate([
+                'note' => 'string',
+            ]);
+            $oldValue = $application->note;
             $application->update(['note' => $request->input('note')]);
-        } elseif ($newStatus) {
+            Mail::bcc($application->committeeMembers())->queue(new ApplicationNoteChanged(user(), $application, $oldValue));
+        }
+        if ($request->has('file')) {
             $this->authorize('editStatus', $application);
-            $application->update(['status' => $newStatus]);
+            $this->storeFile($request, $application->user);
+            Mail::bcc($application->committeeMembers())->queue(new ApplicationFileUploaded($request->get('name'), $application));
```

Comment:
This throws an exception if `committeeMembers` is empty. Please address it post-commit.